### PR TITLE
Convert rz-bin cmd call in rz-find to API

### LINKED ISSUE #1475: Convert `i`/`rz-bin` calls inside Rizin to the API calls
```
librz/main/rz-find.c
409:		rz_sys_cmdf("rizin"
421:			rz_sys_cmdf("rizin -qc \"/E %s\" \"%s\"", kw, efile);

librz/egg/egg_Cfile.c
289:    output = rz_sys_cmd_strf("rz-bin -o '%s.text' -O d/S/'%s' '%s'.o",

librz/core/cmd_debug.c
1673:                           res = rz_sys_cmd_strf("env RZ_BIN_PREFIX=\"%s\" rz-bin %s-B 0x%08" PFMT64x " -S \"%s\" | grep \"%s\"", name, mode, baddr, filesc, sect);
1676:                           res = rz_sys_cmd_strf("env RZ_BIN_PREFIX=\"%s\" rz-bin %s-B 0x%08" PFMT64x " -S \"%s\"", name, mode, baddr, filesc);

```

_Originally posted by @XVilka in https://github.com/rizinorg/rizin/issues/1238#issuecomment-870335117_

<details>
I have removed the following:
```
librz/core/cmd_info.c
703:                            rz_sys_cmdf("rz-bin -O \"%s\" \"%s\"", rz_str_trim_head_ro(input + 1), desc->name);
706:                            rz_sys_cmdf("rz-bin -O help");
```

because those are not `i` commands.
</details>

### PR DESCRIPTION
**Pull request by chinggg**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- [x] Convert `rz-bin -qzzz` call in case `RZ_SEARCH_STRING` to internal API
  - https://github.com/rizinorg/rizin/blob/dev/librz/main/rz-bin.c#L1189
- [ ] `RZ_SEARCH_MAGIC`
- [ ] `RZ_SEARCH_ESIL`
...

**Test plan**

...

**Closing issues**

closes #1475
...

### CODE REVIEW

**Review by ret2libc — CHANGES_REQUESTED** [review]
How is `state` used here? Also, is it really necessary to have `core`?

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by chinggg** [comment]
> How is state used here? Also, is it really necessary to have core?

You are right, `state` is not used since the output mode is specified by `bf->strmode`.

If RzCore is not used, more code is needed to make RzBin work, which do you prefer?
```c
RzBin *bin = rz_bin_new();
RzIO *rio = rz_io_new();
rio->cb_printf = rz_cons_printf;
rz_io_bind(rio, &bin->iob);
// ..
rz_bin_free(bin);
rz_io_free(rio);
```

**Comment by wargio** [comment]
you can just printf without rz_cons

**Review by XVilka — APPROVED** [review]

File: librz/main/rz-find.c [hunk 1]
```diff
@@ -294,11 +295,24 @@ static int rzfind_open_file(RzfindOptions *ro, const char *file, const ut8 *data
...
+	rz_core_init(&core);
+	RzBinOptions opt;
+	RzBin *bin = core.bin;
+	rz_bin_options_init(&opt, 0, 0, 0, false, 2);
+	RzBinFile *bf = rz_bin_open(bin, file, &opt);
+	if (ro->json) {
+		bf->strmode = RZ_MODE_JSON;
+	} else {
+		bf->strmode = RZ_MODE_SIMPLE;
+	}
```
Comment (ret2libc) [inline librz/main/rz-find.c]:
```suggestion
	bf->strmode = ro->json ? RZ_MODE_JSON : RZ_MODE_SIMPLE;
```

I think in this case a ternary operator fits very well and it's very readable.

File: librz/main/rz-find.c [hunk 2]
```diff
@@ -294,11 +295,24 @@ static int rzfind_open_file(RzfindOptions *ro, const char *file, const ut8 *data
 		goto err;
 	}
 
+	RzCore core = { 0 };
+	rz_core_init(&core);
+	RzBinOptions opt;
+	RzBin *bin = core.bin;
```
Comment (ret2libc) [inline librz/main/rz-find.c]: I think it's alright to avoid RzCore and do RzBin+RzIO, if necessary.

**Review by ret2libc — APPROVED** [review]
Thanks!

Damn, `rz_bin_dump_strings` API is really bad...

**Comment by chinggg** [comment]
Yes. In fact I think sometimes it is more convenient to use command especially if the API is not well-designed. 
For the other usage of `rz_sys_cmdf` in the same function `rzfind_open_file`, should it be replaced by internal API?

**Comment by wargio** [comment]
> Thanks!
> 
> Damn, `rz_bin_dump_strings` API is really bad...

i will remove it in one of my PRs on RzPrint

**Pull request merged**